compiler/rustc_parse/src/utils.rs

```diff
@@ -0,0 +1,60 @@
+/// copied from rustc_lint
+
+/// Some unicode characters *have* case, are considered upper case or lower case, but they *can't*
+/// be upper cased or lower cased. For the purposes of the lint suggestion, we care about being able
+/// to change the char's case.
+fn char_has_case(c: char) -> bool {
+    let mut l = c.to_lowercase();
+    let mut u = c.to_uppercase();
+    while let Some(l) = l.next() {
+        match u.next() {
+            Some(u) if l != u => return true,
+            _ => {}
+        }
+    }
+    u.next().is_some()
+}
+
+pub fn to_camel_case(s: &str) -> String {
```

Comment:
https://github.com/rust-lang/rust/blob/dcd5177fd43d5fdd9b89a7700fe3c283a9a52a48/compiler/rustc_lint/src/nonstandard_style.rs#L90

Reply:
but it's in crate `rustc_lint`. I think it's weird to make `rustc_parse` depend `rustc_lint`. Should we create a crate `rustc_utils` for these functions like `to_camel_case`

Reply:
We could move this to a shared dependency of the two, like `rustc_errors`.